packages/aws-cdk-lib/aws-ecr/lib/repository.ts

```diff
@@ -361,8 +366,33 @@ export abstract class RepositoryBase extends Resource implements IRepository {
    * Grant the given identity permissions to use the images in this repository
    */
   public grantPull(grantee: iam.IGrantable) {
-    const ret = this.grant(grantee, 'ecr:BatchCheckLayerAvailability', 'ecr:GetDownloadUrlForLayer', 'ecr:BatchGetImage');
+    const ret = this.grant(grantee,
+      'ecr:BatchCheckLayerAvailability',
+      'ecr:GetDownloadUrlForLayer',
+      'ecr:BatchGetImage',
+    );
+
+    iam.Grant.addToPrincipal({
+      grantee,
+      actions: ['ecr:GetAuthorizationToken'],
+      resourceArns: ['*'],
+      scope: this,
+    });
 
+    return ret;
+  }
+
+  /**
+   * Grant the given identity permissions to use the images in this repository
+   */
+  public grantPush(grantee: iam.IGrantable) {
```

Comment:
This feels reasonable to me but I don't have too much context in ecr so just going to make this comment here. 

If we go this route, remember to change `grantPullPush` to call `grantPush` -- there's no reason to duplicate the code in both functions.

But then we also have to audit `ecr:BatchCheckLayerAvailability`, because that's not part of the permissions for `grantPullPush`. Why is it part of `grantPush`? And if it is necessary, it should be a part of `grantPullPush` as well.

Reply:
There was an issue with the 'getauthorizationtoken' being duplicated when executing 'grantPull' and 'grantPush' within 'grantPullPush'. I have extracted the other Actions into S3 Constructs and ensured readability by commonizing them. I would like to discuss whether this answer is appropriate. I am not sure about the details of 'BatchCheckLayerAvailability', but it is an Action in accordance with the official documentation below. I have made it compatible with both 'Pull' and 'PullPush'. 
https://docs.aws.amazon.com/ja_jp/AmazonECR/latest/userguide/image-push.html#image-push-iam

Reply:
I'd like to confirm if the duplication is really an issue here. `iam.Grant` is wise enough to get rid of any duplication, e.g. even if you allow `ecr:GetAuthorizationToken` twice, the policy is synthesized without any error and it has just a single allow policy. Afaik it should be valid and safe to call `grantPull` and `grantPush` in `grantPullPush`.